lib/odometry-ng/include/odometer/OdometerInterface.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+
+#include "cogip_defs/Polar.hpp"
+#include "cogip_defs/Pose.hpp"
+
+namespace cogip {
+
+namespace odometer {
+
+class OdometerInterface {
+
+// Internal structures for easier Position and Velocity management
+struct Position {
+    Position() : x(0), y(0), O(0) {}
+    Position(float x, float y, float O) : x(x), y(y), O(O) {}
+
+    float x, y, O;
+};
+
+struct Velocity {
+    Velocity() : linear(0), angular(0) {}
+    Velocity(float linear, float angular) : linear(linear), angular(angular) {}
+
+    float linear, angular;
+};
+
+public:
+    /// @brief Set the default odometry position
+    ///
+    /// @note this function should be called to reset robot position and defined a new default one
+    ///
+    /// @param x X coordinate (mm)
+    /// @param y Y coordinate (mm)
+    /// @param O angle (deg)
+    virtual void set_position(double x, double y, double O) = 0;
+
+    /// @brief Set the default odometry position
+    ///
+    /// @note this function should be called to reset robot position and defined a new default one
+    ///
+    /// @param pose postion reference
+    virtual void set_position(const cogip::cogip_defs::Pose &pose) = 0;
+
+    /// @brief Copy current position using cogip def format
+    /// @note Data units:
+    ///         - x, y: mm
+    ///         - O: deg
+    /// @param pose cogip::cogip_defs::Pose current pose
+    virtual void copy_position(cogip::cogip_defs::Pose &pose) = 0;
```

Comment:
Fixed for typo.

I tried this but this is impossible due to the current integration inside motion control.

In `motion_control.cpp`, the function `void compute_current_speed_and_pose(cogip::cogip_defs::Polar &current_speed,
                                    cogip::cogip_defs::Pose &current_pose)` give two reference for sped and pose for internal use. 
We need to update the values ans not the reference. 

This is more user friendly to provide a copy method to achieve this instead of doing this in this function.

```c++
void compute_current_speed_and_pose(cogip::cogip_defs::Polar &current_speed,
                                    cogip::cogip_defs::Pose &current_pose)
{
    if(odometry.update() < 0)
    {
        std::cout << "Odometry update failure" << std::endl;
        return;
    }
    
    odometry.copy_position(current_pose);
    odometry.copy_velocities(current_speed);
}
```

Reply:
Replaced by two const reference getter for pose and polar